client/type_helpers.go

```diff
@@ -46,3 +46,23 @@ func Equivalent[T any](a, b []T) bool {
 
 	return true
 }
+
+func IsZero[T comparable](v T) bool {
+	return v == *new(T)
+}
+
+func PtrValueOrDefault[T any](v *T, d T) T {
+	var result = d
+	if v != nil {
+		result = *v
+	}
+	return result
```

Comment:
Do you need the intermediate here (and below)?
```suggestion
	if v != nil {
		return *v
	}
	return d
```

Reply:
Ah! I don't: I think that was leftover from a previous iteration.